hammock/views.py

```diff
@@ -115,3 +116,111 @@ def destroy(self, request, *args, **kwargs):
         """
         return super(AtomicNonIdempotentActionViewSetMixin, self).destroy(
             request, *args, **kwargs)
+
+
+class NestedModelViewSetMixin(object):
+    """Mixin for `ModelViewSet` that is nested under another `ModelViewSet`."""
+
+    # The model class of the nesting `ModelViewSet`
+    # e.g. `User` if the User model is the nesting model
+    nesting_model = None
+
+    # The field name of the nested model that references the parent model
+    # e.g. `user` if the nested or related model's field to the `User`
+    # model is `user`
+    nesting_model_field_name = None
+
+    # The queryset filter keyword argument for matching the parent
+    # model's primary key value
+    # e.g. `user__pk`
+    nesting_model_lookup_field = None
+
+    # The keyword argument in the URL configuration for the parent model
+    nesting_model_lookup_url_kwarg = None
+
+    def get_nesting_model(self):
+        """Return the parent model class."""
+        if self.nesting_model is None:
+            raise ImproperlyConfigured(
+                '`nesting_model` is not set.  You either need to set '
+                '`nesting_model` or implement `get_nesting_model` method.')
+
+        return self.nesting_model
+
+    def get_nesting_model_field_name(self):
+        """Return the field name referencing the related parent model."""
+        if not self.nesting_model_field_name:
+            return self.get_nesting_model()._meta.model_name
+
+        return self.nesting_model_field_name
+
+    def get_nesting_model_lookup_field(self):
+        """Return the queryset filter keyword argument for parent model pk."""
+        if not self.nesting_model_lookup_field:
+            return '{}__pk'.format(self.get_nesting_model()._meta.model_name)
+
+        return self.nesting_model_lookup_field
+
+    def get_nesting_model_lookup_url_kwarg(self):
+        """Return the keyword argument in the URL conf for the parent model."""
+        if not self.nesting_model_lookup_url_kwarg:
+            return '{}_pk'.format(self.get_nesting_model()._meta.model_name)
+
+        return self.nesting_model_lookup_url_kwarg
+
+    def get_nesting_model_instance(self):
+        """Return the instance of the parent model."""
+        return self.get_nesting_model().objects.get(
```

Comment:
@lemuelf @arnoldchoa91 we have issues where the parent model id that was passed is 0 or i could be any number that does not exist. Can we make this use `get_object_or_404` instead?

https://github.com/Zbooni/zbooni-backend/issues/1206

Reply:
Sure. I'm able to replicate it on my end. Will fix.

Reply:
My beef with `get_object_or_404` is it's not very reusable in class-based views (it was a shortcut meant for function-based views, and in my opinion, it doesn't fit very well in class-based views' aim of reusability).  That is why I usually override the `get_object` method to do just that—return the object or `None`, or, in some cases, let it just raise `DoesNotExist`*—and then leave the responsibility of returning `HttpResponseNotFound` in some other method where it makes sense.

In this particular method, though, since it's named `get_nesting_model_instance`, I think it should also do just that: return the model instance or `None` if there really is no nesting model instance (I'm not sure when this would be appropriate but maybe if the mixin is mistakenly used in a non-nested view?  But maybe raising an exception is better for that, though.), or raise a `ValueError` if the values it needs to perform its job is invalid (like an empty string for ID), or `DoesNotExist`, which I don't like but just to be consistent with Django's conventions, if the model instance doesn't really exist.

If we are going to return an `HttpResponseNotFound`**, I think it should be done in the view methods that are expected to return a response, i.e. one of the HTTP method handler methods (`get`, `post`, etc. methods) or the `dispatch` method (as a "catch-all" method for the HTTP method handlers).

* I don't like the `DoesNotExist` exception either; I think it's lazy programming.  But, hey, Django is a framework for making web development shortcuts, so maybe it has a place in it 🤷‍♂️
** raising an Http404 exception is lazy either; I prefer explicitly returning an `HttpResponseNotFound` response